File: LeetCodeExample.Test/Stack/_00946_ValidateStackSequences.py
# Given two integer arrays pushed and popped each with distinct values, 
# return true if this could have been the result of a sequence of push and pop operations on an initially empty stack, or false otherwise.

class Solution:
    def validateStackSequences(self, pushed: List[int], popped: List[int]) -> bool:
        pushStack = []
        n = 0
        #print(popStack)
        for x in pushed:
            pushStack.append(x)
            #print("pushing " + str(pushStack[-1]) + " checking against " + str(popStack[-1]))
            while pushStack and pushStack[-1] == popped[n]:
                #print("popping " + str(pushStack[-1]))
                pushStack.pop()
                n = n + 1
        return not pushStack

File: LeetCodeExample.Test/Stack/_00946_ValidateStackSequences.cs
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Linq;

namespace LeetCodeExample.Test
{
    /// <summary>
    //Given two integer arrays pushed and popped each with distinct values, return true if this could have been the result of a sequence of push and pop operations on an initially empty stack, or false otherwise.
    /// </summary>
    public class _00946_ValidateStackSequences
    {
        public bool ValidateStackSequences(int[] pushed, int[] popped)
        {
            Stack<int> stack = new Stack<int>();
            int poppedIndex = 0;
            for (int i = 0; i < pushed.Length; i++)
            {
                stack.Push(pushed[i]);
                while (stack.Count != 0 && stack.Peek() == popped[poppedIndex])
                {
                    stack.Pop();
                    poppedIndex++;
                }
            }
            return stack.Count == 0;
        }
    }
}